### PR TITLE
Mobile auth and integration for SAML2 SSO

### PR DESCRIPTION
**Pull request by dwang159**

### CODE REVIEW

File: src/iris_relay/app.py [hunk 1]
```diff
@@ -3,23 +3,33 @@
...
 from streql import equals
 from twilio import twiml
 from urllib3.exceptions import MaxRetryError
 import yaml
 import falcon
 import ujson
 import falcon.uri
+import os
+
+from saml2 import (
```
Comment (jrgp) [inline src/iris_relay/app.py]: why not just `from saml2 import entity` ?

File: src/iris_relay/app.py [hunk 2]
```diff
@@ -66,6 +76,101 @@ def compute_signature(token, uri, post_body, utf=False):
...
+                                      %s)
+                              ''',
+                           (username, refresh_token, exp))
+            connection.commit()
+            key_id = cursor.lastrowid
+        finally:
+            cursor.close()
+            connection.close()
+        location = form_data.get('RelayState', self.redirect_url)
+        resp.set_header('Location', ''.join([location, '#token=', refresh_token,
```
Comment (jrgp) [inline src/iris_relay/app.py]: should `#token` really be `?token` ?

Comment (dwang159) [reply src/iris_relay/app.py]: No, I'm using '#' to pass some info to the Javascript. '?' would probably work too, but I don't think there's much difference.

Comment (jrgp) [reply src/iris_relay/app.py]: That's what I wanted to confirm, url hash or query string. Makes sense.

**Pull request merged**